localstack/services/secretsmanager/secretsmanager_listener.py

```diff
@@ -21,7 +21,10 @@ def forward_request(self, method, path, data, headers):
             if parts[3] != aws_stack.get_region():
                 LOG.info('Unexpected request region %s for secret "%s"' % (aws_stack.get_region(), secret_id))
             # secret ARN ends with "-<randomId>" which we remove in the request for upstream compatibility
-            data['SecretId'] = parts[-1].rpartition('-')[0]
+            if parts[-1][len(parts[-1]) - 6: len(parts[-1]) - 5] == '-':
```

Comment:
nit: we can access the character directly via its index (no need to create a range of length 1).

Also, we have 6 occurrences of `parts[-1]` - perhaps would be better for readability if we extract it to a separate local variable:
```
secret_id = parts[-1]
if secret_id[-6] == '-':
    ...
```

Finally - can we be certain that the dash will always be at the index position `[-6]`? What if the random string suffix is more than 5 characters long? I think the logic would be more robust if we `search` or `split` for the `-` contained in `secret_id`, rather than assuming a fixed length. Thoughts?

Reply:
Yeah, I have already linked the issue here. So according to the AWS documentation here after the hyphen a six digit random string is added but in case of Moto a five digit random string is being added.

So in our case the sixth character from the end should be  '-'  . So if that is not the case then we consider the partial Arn has been given and do not remove 

`If you specify an ARN, we generally recommend that you specify a complete ARN. You can specify a partial ARN too—for example, if you don’t include the final hyphen and six random characters that Secrets Manager adds at the end of the ARN when you created the secret. A partial ARN match can work as long as it uniquely matches only one secret. However, if your secret has a name that ends in a hyphen followed by six characters (before Secrets Manager adds the hyphen and six characters to the ARN) and you try to use that as a partial ARN, then those characters cause Secrets Manager to assume that you’re specifying a complete ARN. This confusion can cause unexpected results. To avoid this situation, we recommend that you don’t create secret names ending with a hyphen followed by six characters.
`